generated_code/aderdg.py

```diff
@@ -124,15 +124,10 @@ def __init__(self, order, multipleSimulations, matricesDir):
 
     self.db.update(project2nFaceTo3m)
 
-    selectVelocitySpp = np.zeros((self.numberOfQuantities(), 3))
-    selectVelocitySpp[6:9,0:3] = np.eye(3)
+    selectVelocitySpp = self.mapToVelocities()[:,:3]
     self.selectVelocity = Tensor('selectVelocity', selectVelocitySpp.shape, selectVelocitySpp, CSCMemoryLayout)
 
-    self.selectTractionSpp = np.zeros((self.numberOfQuantities(), 3), dtype=bool)
-    self.selectTractionSpp[0,0] = True
-    self.selectTractionSpp[3,1] = True
-    self.selectTractionSpp[5,2] = True
-
+    self.selectTractionSpp = self.mapToTractions()
```

Comment:
Apologies; I forgot that one here (and didn't expect it to fail).

Once you also set `self.mapToTractions()[:,:3]`, we should be good to go.